VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,415 @@ Function Install-EsxiCertificate {
+                "organizationUnit": "'+ $organizationUnit + '",
+                "state": "'+ $stateOrProvince + '"
+                },
+            '
+            $resourcesBodyObject += [pscustomobject]@{
+                resources = $resourcesObject
+            }
+            $resourcesBodyObject | ConvertTo-Json -Depth 10 | Out-File -FilePath $path"temp.json"
+            Get-Content $path"temp.json" | Select-Object -Skip 1 | Set-Content $path"temp1.json"
+            $resouresJson = Get-Content $path"temp1.json" -Raw
+            Remove-Item -Path $path"temp.json"
+            Remove-Item -Path $path"temp1.json"
+            $requestCsrSpecJson = $csrGenerationSpecJson + $resouresJson
+            $requestCsrSpecJson | Out-File $path"$($workloadDomain)-requestCsrSpec.json"
+            Write-Output "Requesting CSRs for Components Associated with Domain $($workloadDomain)"
+            $myTask = Request-VCFCertificateCSR -domainName $($workloadDomain) -json $path"$($workloadDomain)-requestCsrSpec.json"
+            Do {
+                Write-Output "Waiting for CSRs to be Created for Components Associated with Domain $($workloadDomain)"
+                Start-Sleep 6
+                $response = Get-VCFTask $myTask.id
+            } While ($response.status -eq "IN_PROGRESS")
+            If ($response.status -eq "FAILED") {
+                Write-Output "Workflow Completed with status $($response.status)" 
+            }
+            elseIf ($response.status -eq "SUCCESSFUL") {
+                Write-Output "Workflow Completed with status $($response.status)"
+            } else {
+                Write-Output "Workflow Completed with an unrecognized status of $($response.status). Please check before proceeding."
+            }
+            Write-Output "CSRs for Components Associated with Domain $($workloadDomain) Created"
+        } else {
+            Write-Error "cannot authenticate to SDDC Manager"
+        }
+    } else {
+        Write-Error "cannot connect to SDDC Manager"
+    }
+}
+
+Function Invoke-GenerateCertificates {
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $workloadDomain
+    )
+
+    if (Test-VCFConnection -server $server) {
+        if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
+            $domainType = Get-VCFWorkloadDomain -name $workloadDomain
+            $resourcesObject = gatherSddcInventory -domainType $domainType.type -workloadDomain $workloadDomain
+            Write-Output "Populating $($workloadDomain)-requestCertificateSpec.json"
+            $path = Join-Path $pwd ""
+            $caTypeJson = '{
+                "caType": "Microsoft",
```

Comment:
Should the `caType` be a `ValidateSet` that would allow this to be used with either type?

Reply:
currently we only offer configuration to a Microsoft CA.  I will open an issue on additional support for openssl